README.md

```diff
@@ -44,6 +44,23 @@ Alongside the newly built binary a file called `developer_overrides.tfrc` will b
 back details for setting the `TF_CLI_CONFIG_FILE` environment variable that will enable Terraform to use your locally built provider binary.
 * HashiCorp - [Development Overrides for Provider developers](https://www.terraform.io/docs/cli/config/config-file.html#development-overrides-for-provider-developers). 
 
+Alternatively, you can run the provider directly and tell Terraform how to communicate it with an environment variable.
+This is useful for attaching a debugger like [delve](https://github.com/go-delve/delve) to the provider.
+To do this, make sure the provider is built and present in the local `bin` directory (see above).
+Then, run the provider executable directly with the `--debug` flag.
```

Comment:
To your first question: I wouldn't say it's preferred, just that they have different purposes. The `dev_overrides` method is probably best for general purpose testing and exercising of the plugin as it more closely matches typical Terraform use. This second method is good for debugging mainly (hard to do otherwise). There is some overlap though, so maybe it's confusing to mention them together? We could add a `### Debugging`-type section to make it clear that `dev_overrides` is the primary mechanism?

I probably haven't explained this well enough, maybe I should include commands actually using delve to make it clearer? Basically you run the plugin through delve completely normally (`dlv exec ./bin/tf-prov-fastly -- --debug` or something), which will output some logs to stdout, including the line with the env var. You then copy paste it into another shell, export it, and run `terraform apply` completely normally. The `terraform` binary itself will read the env var and go find the unix socket itself instead of running the plugin in a subprocess as it normally would. Does that make more sense? I can get it working with delve TUI and update the instructions to reflect that if it helps